thold_functions.php

```diff
@@ -2388,157 +2416,167 @@ function thold_check_threshold(&$thold_data) {
 			}
 
 			if ($notify) {
-				$trigger = false;
+				$etrigger = false;
 			} else {
-				$trigger = true;
+				$etrigger = true;
 			}
 
-			$subject = get_email_subject('WARNING', $trigger, $lastread, $ra, $warning_breach_up, $thold_data);
+			$subject = get_email_subject('WARNING', $etrigger, $lastread, $ra, $warning_breach_up, $thold_data);
 
 			if ($notify) {
 				if (!$suspend_notify && !$maint_dev) {
 					thold_debug('Alerting is necessary');
 
-					if ($syslog) {
```

Comment:
This whole bit below this change is pretty meaty.  So, for me to review, it's best to merge.  However, we need to resolve the comments first.  Then, we should really have a state diagram for this flow that also comprehends the "new" method that was added.  Keep in mind that there are two modes.

1) Where the warning and alerts go to the same group, they are the same team and level
2) Where the warnings and alerts go to the different people, and the people who receive the warnings also receive the alerts.

So, technically, there needs to be a State Diagram for both modes.  It really begs for @cigamit original thold 5.0 design that separated triggering from the notification events.

Reply:
Most of the code was not changed but was moved so that the Alert to Warning occurred.  A compare function that shows block moves would be ideal.